File: examples/app-java/src/main/java/pers/xuankai/udptestjava/activity/UDPActivity.java
package pers.xuankai.udptestjava.activity;

import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;
import com.dylanc.longan.SystemBarsKt;
import com.gitee.xuankaicat.kmnkt.socket.ISocket;
import com.gitee.xuankaicat.kmnkt.socket.UDP;
import com.gitee.xuankaicat.kmnkt.socket.utils.CharsetUtils;
import pers.xuankai.udptestjava.BaseActivity;
import pers.xuankai.udptestjava.databinding.ActivityMainBinding;

import java.nio.charset.Charset;

public class UDPActivity extends BaseActivity<ActivityMainBinding> {
    private final UDP socket = (UDP) ISocket.getUDP(s -> {
        s.setAddress("10.0.2.2");
        s.setPort(9000);
        CharsetUtils.setInCharset(s, Charset.forName("gb2312"));
        CharsetUtils.setOutCharset(s, Charset.forName("gb2312"));
        return s;
    });

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        SystemBarsKt.immerseStatusBar(this, true, true);

        ActivityMainBinding binding = getBinding();

        binding.btn.setOnClickListener(v -> {
            String sendText = binding.editText.getText().toString();
            if(sendText.equals("")) return;

            socket.send(1883, sendText);
            binding.textView.setText("等待数据...");
            socket.startReceive((result, ignore) -> {
                binding.textView.setText(result);
                return false;
            });
        });

        binding.btnTcp.setOnClickListener(v -> {
            Intent intent = new Intent(UDPActivity.this, TCPActivity.class);
            startActivity(intent);
        });

        binding.btnMQTT.setOnClickListener(v -> {
            Intent intent = new Intent(UDPActivity.this, MQTTActivity.class);
            startActivity(intent);
        });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        socket.stopReceive();
        socket.close();
    }
}

File: examples/app/src/main/java/pers/xuankai/udptest/activity/UDPActivity.kt
package pers.xuankai.udptest.activity

import android.os.Bundle
import com.dylanc.longan.immerseStatusBar
import com.dylanc.longan.startActivity
import com.gitee.xuankaicat.kmnkt.socket.dsl.udp
import pers.xuankai.udptest.BaseActivity
import pers.xuankai.udptest.databinding.ActivityMainBinding
import java.nio.charset.Charset

class UDPActivity : BaseActivity<ActivityMainBinding>() {
    private val socket = udp {
        address = "10.0.2.2"
        port = 9000
        inCharset = Charset.forName("gb2312")
        outCharset = Charset.forName("gb2312")
        open()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        immerseStatusBar()

        binding.btn.setOnClickListener {
            val sendText = binding.editText.text.toString()
            if(sendText.isEmpty()) return@setOnClickListener

            socket.send(1883, sendText)
            binding.textView.text = "等待数据..."
            socket.startReceive { str, _ ->
                binding.textView.text = str
                return@startReceive false
            }
        }

        binding.btnTcp.setOnClickListener {
            startActivity<TCPActivity>()
        }

        binding.btnMQTT.setOnClickListener {
            startActivity<MQTTActivity>()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        socket.stopReceive()
        socket.close()
    }
}